liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -1136,10 +1255,32 @@ private MavenProject resolveMavenProject(File buildFile) {
         return currentProject;
     }
 
-    private void runTestMojo(String groupId, String artifactId, String goal, MavenProject project) throws MojoExecutionException {
+    private void runTestMojo(String groupId, String artifactId, String goal, MavenProject project)
+            throws MojoExecutionException {
         Plugin plugin = getPluginForProject(groupId, artifactId, project);
         Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(plugin, goal, log);
 
+        // check if this is a project module or main module
+        if (util.isMultiModuleProject()) {
+            try {
+                Set<String> testArtifacts;
+                ProjectModule projectModule = util.getProjectModule(project.getFile());
+                if (projectModule != null) {
+                    testArtifacts = projectModule.getTestArtifacts();
+                } else {
+                    // assume this is the main module
+                    testArtifacts = util.getTestArtifacts();
+                }
+                if (goal.equals("test") || goal.equals("integration-test")) {
+                    injectClasspathElements(config, testArtifacts, project.getTestClasspathElements());
+                }
+            } catch (IOException | DependencyResolutionRequiredException e) {
+                log.warn(
+                        "Unable to resolve test artifact paths, test compilation may not include parent dependencies. Restart dev mode to ensure classpaths are properly resolved.");
```

Comment:
From the message, "test compilation may not include parent dependencies" does not make it clear whether the user did something wrong.  Also, suggest to use `error` logging to indicate that dev mode had an error.